server/main.py

```diff
@@ -78,18 +81,19 @@ async def retrieve_file(key: str, metadata_only: Optional[bool] = False):
 
 @app.put("/update/")
 async def update_files(key: str, encoded_content: List[str] = Form(...), new_key: Optional[str] = Form(None)):
-    filepath = get_metadata(key)
 
-    if filepath is None:
-        raise HTTPException(status_code=404, detail="Key not found")
+    if not check_key_existence(key):
+        await upload(key, encoded_content)
+        return JSONResponse(content={"message": "Files updated successfully"}) 
```

Comment:
think about this. When you call upload, upload is already returning something to you. Where are you capturing this?
You should ideally do `await return upload(key, encoded_content)`. You are returning without a key during update?